### ISSUE TITLE
Integration Test with Nested Components Only Renders Outermost Component

### ISSUE THREAD

**Issue by kkincade** [emberjs/ember.js#14849]
I upgraded my application to Ember 2.10.2 and have come across broken tests due to components not being rendered. It seems to be specifically integration tests where the component uses another component inside itself.

For example, my integration test looks like the following:

```
import { moduleForComponent, test } from 'ember-qunit';
import hbs from 'htmlbars-inline-precompile';
import loadAppState from '../../../../../../util/load-app-state';

moduleForComponent('modals/automation/rules/new-image-sensor-rule', 'Integration | Component | modals/automation/rules/new image sensor rule', loadAppState({
    integration: true
}));

test('it renders', function (assert) {
    this.render(hbs`{{modals/automation/rules/new-image-sensor-rule}}`);

    assert.ok(this.$('.new-rule-dialog.modal-lg').length, 'Modal renders');
    assert.equal(this.$('li').length, 6, 'List of suggested image sensor rules renders');
});
```

The **new-image-sensor-rule** component renders another component I have defined called **simple-dialog**.

```
<div>For testing purposes.</div>
{{#modals.modal-dialog.simple-dialog buttons=buttons title=title target=this class=dialogClassNames}}
    <ul class="buttons collection">
        {{#each smartRules as |rule|}}
            <li>
                {{button-adc icon="add" text=rule.title primary=true on-click=(action 'createRule' rule)}}
                <p>{{rule.description}}</p>
            </li>
        {{/each}}
    </ul>

    <h4>{{tc this 'suggested'}}</h4>

    <ul class="alerts collection">
        {{#each basicRules as |rule|}}
            <li {{action 'createRule' rule}}>
                <h5>{{rule.title}}</h5>
                <p>{{rule.description}}</p>
                {{svg-symbol icon="circle-add"}}
            </li>
        {{/each}}
    </ul>
{{/modals.modal-dialog.simple-dialog}}
```

When debugging the test, the only element rendered is the div with "For testing purposes." text inside of it. Is there something special I need to do to test nested components in integration tests in Ember 2.10.x? Or is this a bug?

**Comment by pixelhandler** [comment]
@kkincade I don't think this is a bug, see this ember-twiddle example, https://ember-twiddle.com/ad2366883da545b61916b620a05e81a6?openFiles=tests.integration.components.red-five.js%2Ctemplates.components.red-six.hbs

The `red-five` component has a nested `red-six` component within it's template file. The integration test for the `red-five` component asserts that the `red-six` component is rendered.

**Comment by kkincade** [comment]
I'm not familiar with twiddles, but your tests don't seem to run (at least no assertions are made). I've tried putting "debugger;" lines in the tests, but they are never hit. Is there something I have to do to make the tests run?

![no-tests-run](https://cloud.githubusercontent.com/assets/2199847/22157768/089f76f4-df07-11e6-85b7-17ec9a01d3ae.png)

**Comment by kkincade** [comment]
@rwjblue, @locks, @pixelhandler

I spent a lot of time looking at this and I think I finally figured out the issue. The problem lies in using dot.separated.paths to specify components.

```
// These ways work fine.
{{#component 'path/to/component'}}{{/component}}
{{#component 'path.to.component'}}{{/component}}
{{#path/to/component}}{{/path/to/component}}

// This way does not work.
{{#path.to.component}}{{/path.to.component}}
```

I've replicated this on a fresh app located [here](https://github.com/kkincade/nested-components). There is no deprecation warning/error within the console, so I'm assuming this is a bug. Not only does this affect tests, but it affects actual applications and their ability to render components.

**Comment by kkincade** [comment]
@rwjblue, @locks Is this intended functionality or is this a bug?

**Comment by rwjblue** [comment]
Nested invocation with `.` is definitely not supported (and was not intentional to begin with). We didn't actually realize this worked (and that folks had stumbled upon it) until we had replaced he rendering engine and baked in the idea that period separated paths could only invoke components if they are closure components (sometimes also known as contextual components) and period separated paths will not do lookups.

tldr; I do not consider this a bug, but I am quite sorry it has caused you grief.

**Comment by mixonic** [comment]
Invocation with `.` in a component name instead of `/` for directory dividers was not public API or documented as such. Sorry you were bit by this @kkincade. I appreciate your time reporting the issue.

I'm closing this as a wontfix.
